core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2179,6 +2179,16 @@ private[spark] object Utils extends Logging {
       .getOrElse(UserGroupInformation.getCurrentUser().getShortUserName())
   }
 
+
+  // Returns the groups to which the current user belongs.
+  def getCurrentUserGroups(sparkConf: SparkConf, username: String): Set[String] = {
+    val groupMappingServiceProvider = Utils.classForName(sparkConf.get("spark.user.groups.mapping",
```

Comment:
Also we want this conf to be basically optional so it should be able to set to empty string and not error out in case people don't want it